main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -848,6 +849,70 @@ object CompletionProvider {
     }
   }
 
+  /**
+    * Get completions for java imports.
+    */
+  private def getImportCompletions()(implicit context: Context, root: TypedAst.Root): Iterable[CompletionItem] = {
+    if (root == null) Nil else getImportNewCompletions()
+  }
+
+  /**
+    * Get completions for importing java constructors.
+    */
+  private def getImportNewCompletions()(implicit context: Context): Iterable[CompletionItem] = {
+    val regex = raw"\s*.*import\s+new\s+(.*)".r
+    context.prefix match {
+      case regex(clazz) => {
+        try {
+          val clazzObject = java.lang.Class.forName(clazz)
+          clazzObject.getConstructors().map(constructor => {
+            val types = constructor.getParameters().map(param => convertJavaClassToFlixType(param.getType())).mkString("(", ", ", ")")
+            // Gets the name of the type excluding the package to use as a suggestion for the name of the constructor.
+            val className = clazz.split('.').last
+            val label = s"$clazz$types"
+            val replace = s"$clazz$types: ${convertJavaClassToFlixType(clazzObject)} \\ IO as $${0:new$className};"
+            CompletionItem(
+              label = label,
+              // Prioritise constructors with fewer parameters.
+              sortText = Priority.high(s"${constructor.getParameterCount()}$label"),
+              textEdit = TextEdit(context.range, replace),
+              documentation = None,
+              insertTextFormat = InsertTextFormat.Snippet,
+              kind = CompletionItemKind.Constructor)
+          })
+        }
+        catch {
+          case _: Throwable => Nil
```

Comment:
I don't think we should catch Throwable, we should be more precise.
Also, add a comment to explain what this case corresponds to (i.e. the class was not found).